### PR TITLE
secrets/database: Fixes marshalling bug for json.Number types

### PR DESCRIPTION
**Pull request by austingebauer**
## Overview

This PR fixes an issue that prevents numeric arguments from being provided to parameters such as [max_*_connections](https://www.vaultproject.io/api-docs/secret/databases/oracle#max_open_connections) in the database secrets engine. The Vault CLI provides the arguments as type string, so the error only occurs when using a different client, for example,`curl`. See an example of the error that's returned when providing an integer argument in the testing section below.

## Testing

I tested that integer arguments can be provided to configure the database secrets engine with `curl`. I also wrote a unit test case which exercises the `json.Number` conversion in `mapToStruct()`.

Example of error that's returned for integer arguments:
```
$ curl -X PUT -H "X-Vault-Request: true" -H "X-Vault-Token: $(vault print token)" -d '{"allowed_roles":"*","connection_url":"{{username}}/{{password}}@//localhost:1521/oraclepdb","max_open_connections":10,"max_idle_connections":13,"password":"vaultpwd","plugin_name":"vault-plugin-database-oracle","username":"vaultadmin"}' http://127.0.0.1:8200/v1/database/config/oracle

{"errors":["error creating database object: unable to marshal config: proto: invalid type: json.Number"]}
```

Example of successful request using changes in this PR:
```
$ curl -X PUT -H "X-Vault-Request: true" -H "X-Vault-Token: $(vault print token)" -d '{"allowed_roles":"*","connection_url":"{{username}}/{{password}}@//localhost:1521/oraclepdb","max_open_connections":10,"max_idle_connections":13,"password":"vaultpwd","plugin_name":"vault-plugin-database-oracle","username":"vaultadmin"}' http://127.0.0.1:8200/v1/database/config/oracle

$ vault read -format=json database/config/oracle
{
  "request_id": "bbc881b3-609a-1360-1a28-83ba8aa9fef5",
  "lease_id": "",
  "lease_duration": 0,
  "renewable": false,
  "data": {
    "allowed_roles": [
      "*"
    ],
    "connection_details": {
      "connection_url": "{{username}}/{{password}}@//localhost:1521/oraclepdb",
      "max_idle_connections": 13,
      "max_open_connections": 10,
      "username": "vaultadmin"
    },
    "password_policy": "",
    "plugin_name": "vault-plugin-database-oracle",
    "root_credentials_rotate_statements": []
  },
  "warnings": null
}
```

### CODE REVIEW

**Review by calvn — APPROVED** [review]

**Comment by calvn** [comment]
Can you add a changelog entry for this?

**Comment by austingebauer** [comment]
@calvn - Will do!

File: changelog/11451.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+secrets/database: Fix marshalling to allow providing integer arguments to the `max_open_connections` and `max_idle_connections` fields.
```
Comment (pcman312) [inline changelog/11451.txt:2]:
```suggestion
secrets/database: Fix marshalling to allow providing numeric arguments to external database plugins.
```

**Review by pcman312 — APPROVED** [review]
Just a nit on the changelog entry (since this change affects external DB plugins only)

**Pull request merged**